### PR TITLE
update docs

### PR DESCRIPTION
**Pull request by ankitjena**
Add documentation about generation config, directives and custom generation with examples.

### CODE REVIEW

File: docs/config.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+---
+id: config
+title: CRUD Config
+---
+
+GraphBack takes input model and generates CRUD operations as queries and mutations.
+ These include `create`, `update`, `delete`, `find` and `findAll`. These are generated for each `type` in your model.
```
Comment (wtrocki) [inline docs/config.md]:
```suggestion
 These include `create`, `update`, `delete`, `find` and `findAll`. These operations can be generated for each `type` in your model.
```

File: docs/config.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ Further Graphback also generates three predefined subscriptions, namely `new`, `updated` and `deleted`.
```
Comment (wtrocki) [inline docs/config.md]:
```suggestion
 Further Graphback also generates three predefined subscriptions, namely `new`, `updated` and `deleted`.
 We can use them in clients to receive live updates for every change that is happening on the server.
```

File: docs/config.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    "find": true,
+    "delete": false,
+    "subCreate": false,
+    "subUpdate": false,
+    "subDelete": false
+  }
+}
+```
+Changing these flags and performing `generate`, regenerates your `schema` and `resolvers` with provided config.
+>Note: For subscriptions, the user needs to change value of respective operation to `true`. For example, changing
```
Comment (wtrocki) [inline docs/config.md]:
```suggestion
> Note: For subscriptions, the user needs to change the value of the respective operations to `true`. For example, changing 
```

File: docs/config.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+`subDelete` to `true` won't work unless, `delete` is `true`.
+
+## Directives
+Changing the generator config applies the config to all the types in your schema. Graphback allows you change these for any single type using directives. All the config flags are available as directives - `@create`, `@update`, `@delete`, `@find`, `@findAll`, `@subCreate`, `@subUpdate` and `@subDelete`.
```
Comment (wtrocki) [inline docs/config.md]:
```suggestion
Changing the generator config applies the config to all the types in your schema. Graphback allows you to change these for any single type using directives. All the config flags are available as directives 

- `@create`
- `@update` 
- `@delete` 
- `@find`
- `@findAll`
- `@subCreate`
- `@subUpdate` 
- `@subDelete`
```

File: docs/config.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+User can use these directives to have more control on individual elements. For example, 
+```
+type Note @delete {
+  ...
+}
+```
+will create the `delete` mutation for `Note` type only.
```
Comment (wtrocki) [inline docs/config.md]:
```suggestion
will create only the `delete` mutation for the `Note` type.
```

File: docs/config.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+Changing these flags and performing `generate`, regenerates your `schema` and `resolvers` with provided config.
+>Note: For subscriptions, the user needs to change value of respective operation to `true`. For example, changing
+`subDelete` to `true` won't work unless, `delete` is `true`.
+
+## Directives
+Changing the generator config applies the config to all the types in your schema. Graphback allows you change these for any single type using directives. All the config flags are available as directives - `@create`, `@update`, `@delete`, `@find`, `@findAll`, `@subCreate`, `@subUpdate` and `@subDelete`.
+
+User can use these directives to have more control on individual elements. For example, 
+```
+type Note @delete {
```
Comment (wtrocki) [inline docs/config.md]: Missing: Disabling defaults. I think we should document that directives override config so if you pass delete if will override delete from config, but how we can disable them if the global config is true.

File: docs/config.md (hunk 5 above)
Comment (wtrocki) [reply docs/config.md]: Is that true? I was thinking that we going to use directives to change global flag. 
This is very counter intuitive to what normally people will expect.

File: docs/custom.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  id: ID!
+  name: String!
+}
+```
+and we want to implement a mutation to like a Post, then we can simply
+```graphql
+...
+...
+
+#Custom methods
```
Comment (wtrocki) [inline docs/custom.md]:
```suggestion
#Custom mutations
```

File: docs/custom.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+export const likePost = {
+  Mutation: {
+    likePost: (_: any, args: any, context: GraphQLContext) => {
+        // Implementation here
+    }
+  }
+}
+```
+
+>This works similarily for `Query` and `Subscription`.
```
Comment (wtrocki) [inline docs/custom.md]:
```suggestion
> **Note:** This works similarily for `Query` and `Subscription`.
```

**Comment by wtrocki** [comment]
I have found some issues, but they might be related to implementation. Please check comments and suggestions.

**Comment by wtrocki** [comment]
If this really work as documented we can merge that and then work on improving it in code and changing docs.

File: docs/config.md (hunk 5 above)
Comment (ankitjena) [reply docs/config.md]: generator config applies to all types and directives add certain operation to a single type.

File: docs/config.md (hunk 6 above)
Comment (ankitjena) [reply docs/config.md]: Yes adding that

**Comment by wtrocki** [comment]
Yes. I think we have bug. I have true in config but want to disable this using directive. WDYT?

**Comment by ankitjena** [comment]
> Yes. I think we have new feature here. I have true in config but want to disable this using directive. WDYT?

Use directives to invert the config value?

**Comment by wtrocki** [comment]
Yup.. We can always enable it by adding directive but disabling is not there

**Comment by wtrocki** [comment]
Minor priority though for the moment

**Pull request merged**